### PR TITLE
feat(globe): selection math + fairness draw for the spin gesture

### LINKED ISSUE #127: Country selection math as a tested pure module
## Parent

[\[PRD\] v2.0.0 globe-as-output gesture redesign](https://github.com/taewanu/sounds-abroad/issues/125)

## What to build

Extract the country selection math into a pure, testable module with no React or Three.js side effects, and cover it with unit tests. Three pure functions: front-face projection (which countries are on the camera-facing hemisphere, projected to screen space), nearest-to-point (the tap target), and the fling snap pool (nearest-N by direction). This is the deep module the fling controller will depend on.

Also add a STANDING structural-reachability test: every one of the 63 countries appears in some rest-point's nearest-N pool, so no country is structurally starved. Verified offline (0/63 starved across 24,240 sampled rest points); this test promotes that check into CI so adding a country in a dense cluster fails the build instead of silently degrading coverage.

Rationale: ADR-0011. Logic already validated in the spike (branch `spike/gesture-feel`).

## Acceptance criteria

- [x] Selection math lives in a pure module: no React, no Three.js side effects.
- [x] Unit tests: front-face projection excludes countries behind the limb.
- [x] Unit tests: nearest-to-point returns the closest front-facing country.
- [x] Unit tests: the snap pool is the nearest N countries by direction.
- [x] Standing reachability test: all 63 countries appear in some nearest-N pool (0 starved).

## Blocked by

None - can start immediately (consumed by the fling controller slice)

### PR DESCRIPTION
**Pull request by taewanu**
The pure, tested selection module the v2.0.0 globe-as-output gesture is built on (ADR-0011). No React, no Three.js side effects: inputs in, a country code out.

## What's here
- `projectFrontCountries` — countries on the camera-facing hemisphere, projected to screen space (the tap frame).
- `pickNearestToPoint` — the tap target: the nearest front-facing country to a screen point.
- `nearestPool` — the fling's candidate pool: the nearest N countries to the rest direction.
- `weightedDraw` — the anti-repeat fairness draw: a weighted pick within the pool biased toward not-yet-visited countries. Randomness is injected (the caller passes it), so the bias is unit-tested deterministically.
- `pickSnapCountry` — composes them: fairness-off returns the literal nearest; fairness-on draws from the pool.

## Scope
- Closes #127 — selection math as a tested pure module, plus the standing reachability test.
- Advances #129: this is its fairness algorithm and unit tests. The remaining #129 criteria — the per-session `visited` set lifecycle (resets on reload) and settle updating that set — are wired in the controller slice, so #129 closes there, not here.
- Part of the #125 PRD.

## Test plan
- `pnpm test` — 358 pass, including 10 new tests for this module.
- Standing reachability test sweeps 24,240 rest points (1.5° grid, |el| ≤ 75°) and asserts all 63 countries appear in some nearest-N pool, promoting ADR-0011's offline check into CI.
- `pnpm typecheck`, `pnpm lint`, `pnpm build` — all green locally.

## Summary by CodeRabbit

* **New Features**
  * Improved country selection on interactive globe for tap and drag interactions
  * Added fairness mode to avoid selecting the same country repeatedly in quick succession

* **Tests**
  * Comprehensive test coverage for country selection utilities and edge cases

### CODE REVIEW

**Comment by taewanu** [comment]
✅ Reviewed [`3dae5fcf6b5d`](https://github.com/taewanu/sounds-abroad/commit/3dae5fcf6b5d33c8ff527121a85e4c5dd799079d)

_Scope: [`dc906c5af496..3dae5fcf6b5d`](https://github.com/taewanu/sounds-abroad/compare/dc906c5af49623477bd1057eae19d6c28eff1dae...3dae5fcf6b5d33c8ff527121a85e4c5dd799079d)_

**No findings**

> Clean rng-injection refactor of `pickSnapCountry`, mirroring the existing `r`-injection in `weightedDraw`.
>
> - Default `rng = Math.random` keeps production behavior identical
> - No non-test callers, so the signature change breaks nothing
> - New determinism test correctly pins both ends of the 10-member pool
> - Nothing high-signal to flag

<details><summary>2 files</summary>

- `src/components/globe/spin-select.test.ts`
- `src/components/globe/spin-select.ts`

</details>

🤖 _pr-review-agent_

File: src/components/globe/spin-select.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+export function pickSnapCountry(
+  el: number,
+  az: number,
+  visited: ReadonlySet<string>,
+  fair: boolean,
+): string {
+  const pool = nearestPool(el, az, SNAP_POOL);
+  if (!fair) return pool[0];
+  return weightedDraw(pool, visited, Math.random());
+}
```
[the remark this answers is not in the document]
Comment (taewanu) [reply src/components/globe/spin-select.ts:120]: Addressed in 3dae5fc: `pickSnapCountry` now takes an `rng` parameter (defaulting to `Math.random`), threaded into the `weightedDraw` call. Production behavior is unchanged; a new test pins the fair-mode draw deterministically via an injected `rng`.

**Comment by taewanu** [comment]
@coderabbitai help

**Comment by taewanu** [comment]
@coderabbitai review
